README.md

```diff
@@ -224,6 +207,97 @@ JoliCode\Elastically\Client:
             ...
 ```
 
+### Using Messenger for async indexing
+
+Elastically ships with a default Message and Handler for Symfony Messenger.
+
+Register the message in your configuration:
+
+```yaml
+framework:
+    messenger:
+        transports:
+            async: "%env(MESSENGER_TRANSPORT_DSN)%"
+
+        routing:
+            # async is whatever name you gave your transport above
+            'JoliCode\Elastically\Messenger\IndexationRequest':  async
+
+services:
+    App\Messenger\IndexationRequestHandler: ~
+```
+
+You have to implement `App\Messenger\IndexationRequestHandler` by extending the core `\JoliCode\Elastically\Messenger\IndexationRequestHandler` so you can plug your database or any other source of truth.
+
+Then from your code you have to call:
+
+```php
+use JoliCode\Elastically\Messenger\IndexationRequest;
+use JoliCode\Elastically\Messenger\IndexationRequestHandler;
+
+$bus->dispatch(new IndexationRequest(Product::class, '1234567890'));
+
+// Third argument is the operation, so for a delete:
+// new IndexationRequest(Product::class, 'ref9999', IndexationRequestHandler::OP_DELETE);
+```
+
+And then consume the messages:
+
+```sh
+$ php bin/console messenger:consume async
+```
+
+### Grouping IndexationRequest in a spool
+
+Sending multiple `IndexationRequest` during the same Symfony Request is not always appropriate, it will trigger multiple Bulk operations. Elastically provides a Kernel terminate listener to group all the `IndexationRequest` in a single `MultipleIndexationRequest` message.
```

Comment:
1. it's not a kernel terminate listener. I would simply remove `terminate` 
2. Why naming it `Multiple` and not `Bulk`

Reply:
Naming it `Bulk` refer to the low level Elastica implementation, but at this level, it's just "multiple indexation request", you don't know if it's one, two, three bulk requests...